core/src/main/scala/caliban/Value.scala

```diff
@@ -201,6 +210,106 @@ private object ValueCirce {
     })
 }
 
+private object ValueJsoniter {
+  import com.github.plokhotnyuk.jsoniter_scala.core._
+  import com.github.plokhotnyuk.jsoniter_scala.macros._
+
+  implicit val streamValueCodec: JsonValueCodec[ResponseValue.StreamValue] =
+    new JsonValueCodec[ResponseValue.StreamValue] {
+      def decodeValue(in: JsonReader, default: ResponseValue.StreamValue): ResponseValue.StreamValue =
+        default // TODO should I create a decoder for stream values?
+      def encodeValue(x: ResponseValue.StreamValue, out: JsonWriter): Unit =
+        out.writeVal(x.toString)
+      def nullValue: ResponseValue.StreamValue =
+        null
+    }
+
+  implicit val booleanValueCodec: JsonValueCodec[Value.BooleanValue] =
+    new JsonValueCodec[Value.BooleanValue] {
+      def decodeValue(in: JsonReader, default: Value.BooleanValue): Value.BooleanValue =
+        Value.BooleanValue(in.readBoolean())
+      def encodeValue(x: Value.BooleanValue, out: JsonWriter): Unit =
+        out.writeVal(x.value)
+      def nullValue: Value.BooleanValue =
+        null
+    }
+
+  implicit val stringValueCodec: JsonValueCodec[Value.StringValue] =
+    new JsonValueCodec[Value.StringValue] {
+      def decodeValue(in: JsonReader, default: Value.StringValue): Value.StringValue =
+        Value.StringValue(Option(in.readString(null)).get)
```

Comment:
`Value.StringValue(in.readString(null))` could be here instead... `in.readString(null)` never returns `null`.